worlds/stardew_valley/test/TestRules.py

```diff
@@ -596,6 +597,38 @@ def swap_museum_and_bathhouse(multiworld, player):
     bathhouse_entrance.connect(museum_region)
 
 
+class TestToolVanillaRequiresBlacksmith(SVTestBase):
+    options = {
+        options.EntranceRandomization: options.EntranceRandomization.option_buildings,
```

Comment:
I believe you shouldn't do an entrance randomizer in this test. It's going to be more consistent and reliable if all entrances are vanilla, except for the specific ones you swap.

Reply:
Then how come `TestDonationLogicAll()` works? Shouldn't the museum also have this problematic short circuit?

Reply:
Museum does not have the short circuit problem It does not, because all the locations are actually located in the museum, so the real entrance rule is checked without passing by `region_logic.py`.